api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimService.java

```diff
@@ -42,25 +44,54 @@
             EligibilityStatus.INELIGIBLE, ClaimStatus.REJECTED
     );
 
-    public ClaimResult createClaim(Claimant claimant) {
+    public ClaimResult createOrUpdateClaim(Claimant claimant) {
         try {
             EligibilityAndEntitlementDecision decision = eligibilityAndEntitlementService.evaluateNewClaimant(claimant);
+            if (claimExistsAndIsEligible(decision)) {
+                Claim claim = findAndUpdateClaim(claimant, decision);
+                return createResult(claim, decision.getVoucherEntitlement());
+            }
+
             Claim claim = createAndSaveClaim(claimant, decision);
             if (claim.getClaimStatus() == ClaimStatus.NEW) {
                 sendNewCardMessage(claim, decision);
                 return createResult(claim, decision.getVoucherEntitlement());
             }
+
             return createResult(claim);
         } catch (RuntimeException e) {
             createAndSaveClaim(claimant, buildWithStatus(EligibilityStatus.ERROR));
             throw e;
         }
     }
 
+    private boolean claimExistsAndIsEligible(EligibilityAndEntitlementDecision decision) {
+        return decision.getExistingClaimId() != null && decision.getEligibilityStatus() == EligibilityStatus.ELIGIBLE;
+    }
+
+    private Claim findAndUpdateClaim(Claimant claimant, EligibilityAndEntitlementDecision decision) {
+        Optional<Claim> optionalClaim = claimRepository.findById(decision.getExistingClaimId());
+        Claim claim = optionalClaim.orElseThrow(() -> new IllegalStateException("Unable to find claim with id " + decision.getExistingClaimId()));
+        List<String> updatedFields = updateClaim(claim, claimant);
+        log.info("Updated claim: {},  fields updated {}", claim.getId(), updatedFields);
+        eventAuditor.auditUpdatedClaim(claim, updatedFields);
+        return claim;
+    }
+
+    private List<String> updateClaim(Claim claim, Claimant claimant) {
+        if (!Objects.equals(claim.getClaimant().getExpectedDeliveryDate(), claimant.getExpectedDeliveryDate())) {
+            claim.getClaimant().setExpectedDeliveryDate(claimant.getExpectedDeliveryDate());
+            claimRepository.save(claim);
+            return singletonList("expectedDeliveryDate");
```

Comment:
Can we have an enum for the fieldnames please as they need to be well defined and match up to the UI.

Reply:
Added enum